src/RestApi/StoreApi/Controllers/Cart.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * Cart controller.
+ *
+ * @internal This API is used internally by Blocks--it is still in flux and may be subject to revisions.
+ * @package WooCommerce/Blocks
+ */
+
+namespace Automattic\WooCommerce\Blocks\RestApi\StoreApi\Controllers;
+
+defined( 'ABSPATH' ) || exit;
+
+use \WP_Error as RestError;
+use \WP_REST_Server as RestServer;
+use \WP_REST_Controller as RestContoller;
+use Automattic\WooCommerce\Blocks\RestApi\StoreApi\Schemas\CartSchema;
+use Automattic\WooCommerce\Blocks\RestApi\StoreApi\Utilities\CartController;
+
+/**
+ * Cart API.
+ */
+class Cart extends RestContoller {
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc/store';
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'cart';
+
+	/**
+	 * Schema class instance.
+	 *
+	 * @var array
+	 */
+	protected $cart_schema;
+
+	/**
+	 * Setup API class.
+	 */
+	public function __construct() {
+		$this->cart_schema = new CartSchema();
+	}
+
+	/**
+	 * Register routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			[
+				[
+					'methods'  => RestServer::READABLE,
+					'callback' => [ $this, 'get_item' ],
+					'args'     => [
+						'context' => $this->get_context_param( [ 'default' => 'view' ] ),
+					],
+				],
+				'schema' => [ $this, 'get_public_item_schema' ],
+			]
+		);
+	}
+
+	/**
+	 * Get the cart.
+	 *
+	 * @param \WP_REST_Request $request Full details about the request.
+	 * @return \WP_Error|\WP_REST_Response
+	 */
+	public function get_item( $request ) {
```

Comment:
I was following naming conventions - in this case the 'item' being returned by the endpoint is a cart object. I realise there might be confusion due to 'cart items' being a thing so instead of changing this, ill make sure cart items are always called 'cart_item'.

Reply:
@Aljullu we always should follow this documentation: https://developer.wordpress.org/rest-api/extending-the-rest-api/controller-classes/#extending-internal-classes
This helps keep the principle of single responsibility per endpoint.